code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -189,180 +189,186 @@
+			return TRUE
+		if(item_type == "condimentBottle")
+			var/obj/item/reagent_containers/food/condiment/P
+			for(var/i = 0; i < amount; i++)
+				P = new/obj/item/reagent_containers/food/condiment(drop_location())
 				P.originalname = name
 				P.name = trim("[name] bottle")
-				reagents.trans_to(P, P.volume, transfered_by = usr)
-			else
-				var/amount_full = 0
-				var/vol_part = min(reagents.total_volume, 30)
-				if(text2num(many))
-					amount_full = round(reagents.total_volume / 30)
-					vol_part = reagents.total_volume % 30
-				var/name = stripped_input(usr, "Name:","Name your bottle!", (reagents.total_volume ? reagents.get_master_reagent_name() : " "), MAX_NAME_LEN)
-				if(!name || !reagents.total_volume || !src || QDELETED(src) || !usr.canUseTopic(src, !issilicon(usr)))
-					return
-
-				var/obj/item/reagent_containers/glass/bottle/P
-				for(var/i = 0; i < amount_full; i++)
-					P = new/obj/item/reagent_containers/glass/bottle(drop_location())
-					P.name = trim("[name] bottle")
-					adjust_item_drop_location(P)
-					reagents.trans_to(P, 30, transfered_by = usr)
-
-				if(vol_part)
-					P = new/obj/item/reagent_containers/glass/bottle(drop_location())
-					P.name = trim("[name] bottle")
-					adjust_item_drop_location(P)
-					reagents.trans_to(P, vol_part, transfered_by = usr)
-			. = TRUE
-
-		if("analyze")
-			var/datum/reagent/R = GLOB.name2reagent[params["id"]]
-			if(R)
-				var/state = "Unknown"
-				if(initial(R.reagent_state) == 1)
-					state = "Solid"
-				else if(initial(R.reagent_state) == 2)
-					state = "Liquid"
-				else if(initial(R.reagent_state) == 3)
-					state = "Gas"
-				var/const/P = 3 //The number of seconds between life ticks
-				var/T = initial(R.metabolization_rate) * (60 / P)
-				analyzeVars = list("name" = initial(R.name), "state" = state, "color" = initial(R.color), "description" = initial(R.description), "metaRate" = T, "overD" = initial(R.overdose_threshold), "addicD" = initial(R.addiction_threshold))
-				screen = "analyze"
-				return
-
-		if("goScreen")
-			screen = params["screen"]
-			. = TRUE
-
-
+				reagents.trans_to(P, vol_each, transfered_by = usr)
+			return TRUE
+		return FALSE
+
+	if(action == "analyze")
+		var/datum/reagent/R = GLOB.name2reagent[params["id"]]
+		if(R)
+			var/state = "Unknown"
+			if(initial(R.reagent_state) == 1)
+				state = "Solid"
+			else if(initial(R.reagent_state) == 2)
+				state = "Liquid"
+			else if(initial(R.reagent_state) == 3)
+				state = "Gas"
+			var/const/P = 3 //The number of seconds between life ticks
+			var/T = initial(R.metabolization_rate) * (60 / P)
+			analyzeVars = list("name" = initial(R.name), "state" = state, "color" = initial(R.color), "description" = initial(R.description), "metaRate" = T, "overD" = initial(R.overdose_threshold), "addicD" = initial(R.addiction_threshold))
+			screen = "analyze"
+			return TRUE
+
+	if(action == "goScreen")
+		screen = params["screen"]
+		return TRUE
+
```

Comment:
usage of the switch was cleaner and better matches other places in the codebase imo, I'll leave this up to someone else to bitch if they don't like it because I don't feel that strongly about it.

Reply:
Well, ifs just result in less indentation, and you can throw in more conditions, like `action == 'something' && condi`. Just plain more flexible.

Reply:
Fair.